src/oop/static_modifier/medium/q5/MainTest.java

```diff
@@ -0,0 +1,113 @@
+    ObjectTest testClass;
+    ObjectTest outputClass;
+    String className = "Hat";
+    String staticAttributeName = "companyLogo";
+    String attributeName = "colour";
+    String initialName = "Nike";
+    String initialColour = "Green";
+    String testClassName = "Test";
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String classString = "oop.static_modifier.medium.q5." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.static_modifier.medium.q5." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+        String modificationErrorMessage = "You have modified the provided portions of " + className + " class for %s attribute. Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName, String.class),
+                String.format(modificationErrorMessage, attributeName));
+        assertTrue(testClass.hasModifier(attributeName, "private"),
+                String.format(modificationErrorMessage, attributeName));
+        String[] test = new String[]{"Black", "Blue", "Red", "Brown"};
+        for (int i = 0; i < test.length; i++) {
+            Object[][] arguments = {
+                    {test[i], String.class}
+            };
+            Object classInstance = testClass.createInstance(arguments);
+            _assertEquals(test[i], testClass.getFieldValue(classInstance, attributeName),
+                    String.format(modificationErrorMessage, attributeName));
+        }
+    }
+
+    @Test
+    @Tag("dependency1")
+    public void correctTestClass() throws Throwable {
+        Object classInstance = outputClass.createInstance();
+        String[] methodModifiers = {"public"};
+        Object[][] arguments = {
+                {new String[0], String[].class}
+        };
+        outputClass.callMethod("main", arguments, methodModifiers, classInstance, new Clause[]{
+                new StringLiteral("This is a " + initialName + " that is " + initialColour + " in colour"),
+                new Optional(new StringLiteral(" ")),
+                new Optional(new NewLine())
+        });
+    }
+
+    @Test
+    public void hatClassHasCorrectAttributes() {
+        String incorrectNameError = "Your %s class does not include the %s attribute.";
+        String incorrectVisibilityError = "Your %s attribute does not have the correct visibility modifier.";
+        String incorrectStaticError = "Your %s attribute does not have the static modifier.";
+        assertTrue(testClass.hasField(staticAttributeName, String.class),
+                String.format(incorrectNameError, className, staticAttributeName));
+        assertTrue(testClass.hasModifier(staticAttributeName, "private"),
+                String.format(incorrectVisibilityError, staticAttributeName));
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"),
+                String.format(incorrectStaticError, staticAttributeName));
```

Comment:
I would test in this method that the company name is initialized correctly, not in a separate test.